packages/metadata/src/providers/localFileMetadata.provider.ts

```diff
@@ -0,0 +1,95 @@
+import { existsSync, readFileSync } from "fs";
+import { z } from "zod";
+
+import {
+    Cache,
+    ILogger,
+    Token,
+    TokenType,
+    ZKChainMetadata,
+    ZKChainMetadataItem,
+} from "@zkchainhub/shared";
+
+import { FileNotFound, IMetadataProvider, InvalidSchema } from "../internal.js";
+import { ChainSchema, TokenSchema } from "../schemas/index.js";
+
+export const LOCALFILE_METADATA_PREFIX = "local-metadata";
+
+/**
+ * Represents a local file metadata provider.
+ */
+export class LocalFileMetadataProvider implements IMetadataProvider {
+    /**
+     * Constructs a new instance of the LocalFileMetadataProvider class.
+     * @param tokenJsonPath The path to the token JSON file.
+     * @param chainJsonPath The path to the chain JSON file.
+     * @param logger The logger instance.
+     * @param cache The cache instance.
+     * @throws {FileNotFound} if any of the files is not found.
+     */
+    constructor(
+        private readonly tokenJsonPath: string,
+        private readonly chainJsonPath: string,
+        private readonly logger: ILogger,
+        private readonly cache: Cache,
```

Comment:
we had the exact same discussion with @0xkenj1 jajajaj, we ended up agreeing in make it work like the GithubMetadata as a matter of consistency

Reply:
we can reconsider this 😅

Reply:
I think i didn't consider the case in which u redeploy the chainhub instance, if you keep redis running and you had updated the JSON file, then it will pick the data from redis. (Supposing infinite ttl )

Reply:
we'll go with this option ser, makes sense